ccmlib/scylla_node.py

```diff
@@ -536,51 +537,69 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
         jvm_args = jvm_args + ['--collectd-hostname',
                                '%s.%s' % (socket.gethostname(), self.name)]
 
-        # Let's add jvm_args and the translated args
-
-        args = [launch_bin, '--options-file', options_file, '--log-to-stdout', '1'] + jvm_args + translated_args
+        args = [launch_bin, '--options-file', options_file, '--log-to-stdout', '1']
+
+        # holds all extended args except --smp and --memory
+        # that are handled separately
+        ext_args = OrderedDict()
+
+        MB = 1024*1024
+
+        def process_opts(opts, set_during_test = False):
+            opts_i = 0
+            memory_opt = None
+            while opts_i < len(opts):
+                if opts[opts_i].startswith("--scylla-manager="):
+                    opts_i += 1
+                elif opts[opts_i].startswith('-'):
+                    o = opts[opts_i]
+                    opts_i += 1
+                    if '=' in o:
+                        kv = o.replace('=', ' ', 1).split()
+                        key = kv[0]
+                        val = kv[1]
+                    else:
+                        key = o
+                        vals = []
+                        while opts_i < len(opts) and not opts[opts_i].startswith('-'):
+                            vals.append(opts[opts_i])
+                            opts_i += 1
+                        val = ' '.join(vals)
+                    if key not in ext_args:
+                        if key == '--smp':
+                            self._smp = int(val)
+                            if set_during_test:
+                                self._smp_set_during_test = True
+                        elif key == '--memory':
+                            # set just the local memory_opt for calculating self._mem_mb_per_cpu
+                            # self._memory is calculated below only after processing both
+                            # SCYLLA_EXT_OPTS and jvm_args
+                            memory_opt = self.parse_size(val)
+                            if set_during_test:
+                                self._mem_set_during_test = True
+                        else:
+                            ext_args[key] = val
```

Comment:
nit, instead of capturing and mutating `ext_args`, could let this function return `ext_args` and have something like:

```py
ext_args = (process_opts(os.getenv('SCYLLA_EXT_OPTS', "").split(), set_during_test=False) |
            process_opts(jvm_args, set_during_test=True))
```

Reply:
makes sense